test/test_reductions.py

```diff
@@ -1402,104 +1395,101 @@ def test_dim_reduction(self, device):
         # This won't test for 256bit instructions, since we usually
         # only work on 1 cacheline (512bit) at a time and these
         # examples aren't big enough to trigger that.
-        for dtype in types:
-            x = torch.tensor(example, device=device, dtype=dtype)
-            self.assertEqual(x.sum().item(), 16)
-            self.assertEqual(x.sum(0), torch.tensor([4, 5, 7], dtype=sum_dtype[dtype]))
-            self.assertEqual(x.sum(1), torch.tensor([2, 14], dtype=sum_dtype[dtype]))
-            y = torch.tensor(example, device=device, dtype=sum_dtype[dtype])
-            torch.sum(x, 0, out=y)
-            self.assertEqual(x.sum(0), y)
-
-        # Mean not supported for Int types
-        for dtype in types[:2]:
+        x = torch.tensor(example, device=device, dtype=dtype)
+        self.assertEqual(x.sum().item(), 16)
+        self.assertEqual(x.sum(0), torch.tensor([4, 5, 7], dtype=sum_dtype[dtype]))
+        self.assertEqual(x.sum(1), torch.tensor([2, 14], dtype=sum_dtype[dtype]))
+        y = torch.tensor(example, device=device, dtype=sum_dtype[dtype])
+        torch.sum(x, 0, out=y)
+        self.assertEqual(x.sum(0), y)
+
+        # Mean not supported for Int types, and precision is not good for float16 & bfloat16
```

Comment:
`mean` is now supported for `float16` & `bfloat16` but if these asserts are enable for them, the test fails due to an assert because the error isn't within accepted precision bounds.

Reply:
Let's try overriding the precision with @precisionOverride decorator.